analyzer/tools/build-logger/src/ldlogger-tool-gcc.c

```diff
@@ -388,9 +422,37 @@ int loggerGccParserCollectActions(
     if (strstr(toolName_, cppCompiler[i]))
       lang = CPP;
 
+  LoggerVector extendedArgs;
+  loggerVectorInit(&extendedArgs);
+
   for (i = 1; argv_[i]; ++i)
   {
     const char* current = argv_[i];
+
+    /* Compilers (clang) supports passing extra flags using so-called response
+     * files that you specify with the @file syntax. We need to read the
+     * arguments from the given file and extend the argument list with these
+     * options.
+     */
+    if (current[0] == '@') {
+      const char* resposeFile = current + 1;
+
+      char fullPath[PATH_MAX];
+      if (loggerMakePathAbs(resposeFile, fullPath, 1))
+      {
+        readArgumentsFromFile(fullPath, &extendedArgs);
+        continue;
+      }
+    }
+    else
+    {
+      loggerVectorAdd(&extendedArgs, current);
+    }
+  }
+
+  for (i = 0; i < extendedArgs.size; ++i)
```

Comment:
Now there are two loops. First you collect the arguments to `extendedArgs` either from the file or from `argv_` and in this second loop you iterate `extendedArgs` over again. Couldn't this go in one round? These files provided by "@" don't seem to be that frequent for me, so it will always be an unnecessary vector copy.